pkg/apis/gcp/validation/worker.go

```diff
@@ -189,6 +189,16 @@ func validateDataVolumeConfigs(dataVolumes []core.DataVolume, configs []gcp.Data
 		allErrs = append(allErrs, validateHyperDisk(dataVolumes[idx], configDataVolume)...)
 	}
 
+	volumeNames := map[string]int{}
+	for _, v := range configs {
+		volumeNames[v.Name]++
+	}
+	for k, v := range volumeNames {
+		if v > 1 {
+			allErrs = append(allErrs, field.Duplicate(dataVolumeFldPath, k))
+		}
+	}
+
```

Comment:
Something like:
```
	volumeNames := map[string]int{}
	for _, v := range configs {
		volumeNames[v.Name]++
	}
	for k, v := range volumeNames {
		if v > 1 {
			allErrs = append(allErrs, field.Duplicate(dataVolumeFldPath, k))
		}
	}
```
maybe better ? 

If you want to do it like you tried then duplicate error message might not be very clear (like what is index K is duplicated). Plus the index reported as duplicate value can be confusing.

Secondary the idiomatic go way to have such maps would probably be to use `map[string]struct{}` but that is closer to the set implementation above

Reply:
Also please add a unit test